### ISSUE TITLE
[question] Should CCI recipes begin to support the "new" msvc compiler in settings?

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan-center-index#6155]
Some consumers might want to use `msvc` instead of `Visual Studio` in their profiles.
Maybe we should start to:
- replace things like `self.settings.compiler == "Visual Studio"` by `self.settings.compiler in ["Visual Studio", "msvc"]`
- properly check `compiler.version` if it's msvc (not the same versionning than Visual Studio)
- take into account that if msvc, `settings.runtime` is split in `settings.runtime` and `settings.runtime_type`

**Comment by prince-chrismc** [comment]
Yes ❤️ I'd love this feature!

**Comment by jgsogo** [comment]
Hi! You all know that my thought is that if something is not tested in the CI, it might eventually become broken. On the other hand, we all really appreciate the efforts to support more configurations in recipes (thanks for all those iOS -related fixes) and we will really like to push new configurations... but we need to keep the budget under control.

On our plans, we would like first to remove some configurations for technical reasons (read latest comments here: https://github.com/conan-io/conan-center-index/issues/4761#issuecomment-900195185) and probably some others if they are not being requested (@danimtb will have numbers soon). Although we've always thought about iOS, Android,... it is very true that `msvc` compiler should be also a high priority (ping @SSE4 ).

---

**Comment by prince-chrismc** [comment]
Would it be easier to close/link these issues and have a GitHub discussion thread(s) about this topic? I am not sure it appears to the dame audience but it might be useful

**Comment by SSE4** [comment]
until we have usage stats for the most requesting packages, there are no plans to add new compilers, architectures, operation systems, etc.
the short term plan is to remove configurations new docker images can't handle - see also https://github.com/conan-io/conan-center-index/issues/4761#issuecomment-900195185
after that, with stats, we can decide what to add first, it could be new compiler version (e.g. Visual Studio or GCC or Clang), or new platform (e.g. iOS or Android or Emscripten), or new architecture. that might be even something that even not on our radar (FreeBSD, Solaris, etc.).
I think we will be able to share stats and plans as soon as they are available.

**Comment by Croydon** [comment]
Since the Conan `msvc` compiler is supposed to eventually replace the `Visual Studio` one (because VS is an IDE, not a compiler), the discussion here should not be about "additional configurations", but rather WHEN to make the switch to the new `msvc` Conan model and if for some time CCI is supporting both the ease the transition.

**Comment by jgsogo** [comment]
Now that VS 2022 has been released, and together with the [help of _epochs_](https://github.com/conan-io/conan-center-index/pull/8587) it should take so long to add this new compiler. Conan uses `msvc` as default for VS2022, so the sooner we start to support it, the easier it will be to move those recipes forward and add the new configuration.

**Comment by SpaceIm** [comment]
I have only one concern, is msvc versionning stable now in conan client (I'm still very confused by the last change: 193 for cl 19.30 :s))?

**Comment by SSE4** [comment]
> I have only one concern, is msvc versionning stable now in conan client (I'm still confused by the last change: 193 for cl 19.30 😕)?

no, it's not stable, `msvc` compiler is experimental (per https://docs.conan.io/en/latest/reference/config_files/settings.yml.html?highlight=msvc#msvc) and subject to a breaking changes (and changes in versioning already happened in practice, see https://github.com/conan-io/conan/pull/10057).

> Moreover, it would be nice to provide a table with corresponding versions to help CCI contributors. Even when you know Visual Studio, its versionning is hard to understand

I agree, table is extremely useful. maybe I'll create a PR to docs.
